File: hackerrank/30_day_of_code/day_6_lets_review.js
// https://www.hackerrank.com/challenges/30-review-loop/problem

function processData(input) {
    let even = "",
        odd = "";
    for (let i = 0; i < input.length; i++) {
        if (i % 2 === 0) {
            even += input[i];
        } else {
            odd += input[i];
        }
    }
    console.log(`${even} ${odd}`);
    return;
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    for (i of _input.split("\n").slice(1)) processData(i);
});

File: hackerrank/30_day_of_code/day_6_lets_review.py
# https://www.hackerrank.com/challenges/30-review-loop/problem
# Enter your code here. Read input from STDIN. Print output to STDOUT
count = int(input())
input_list = [input() for i in range(count)]


def rev(s):
    ev, od = '', ''
    for i in range(len(s)):
        if i % 2 == 0:
            ev += s[i]
        else:
            od += s[i]
    return f"{ev} {od}"


if __name__ == '__main__':
    for s in input_list:
        print(rev(s))
